### PR TITLE
DOC: Answer two common questions.

### PR DESCRIPTION
**Pull request by danielballan**
Answer two common questions.

Rendered here for easier reading:

image

image

### CODE REVIEW

File: doc/source/plans.rst [hunk 1]
```diff
@@ -150,6 +181,46 @@ pseudo-axis. It's all the same to the plans. Examples:
...
+   .. code-block:: python
+
+      motor.settle_time = 1
+      temperature_controller.settle_time = 10
+
+   For many cases, this is more convenient and more robust than typing a delay
+   parameter in every invocation of the scan. You only have to set it once, and
+   it applies thereafter.
+
+   This is why bluesky leaves ``delay`` out of the scans, to guide users toward
```
Comment (mrakitin) [inline doc/source/plans.rst]: Checking my English, why not `towards`?
```suggestion
   This is why bluesky leaves ``delay`` out of the scans, to guide users towards
```

File: doc/source/plans.rst [hunk 2]
```diff
@@ -150,6 +181,46 @@ pseudo-axis. It's all the same to the plans. Examples:
...
+   right tool for the job, it is of course always possible to add a ``delay``
+   parameter yourself by writing a custom plan. Here is one approach, using a
+   :ref:`per_step hook <per_step_hook>`.
+
+   .. code-block:: python
+
+      def scan_with_delay(*args, delay=0, **kwargs):
+
+          def one_nd_step_with_delay(detectors, step, pos_cache):
+              "Insert a sleeep after each step."
```
Comment (mrakitin) [inline doc/source/plans.rst]:
```suggestion
              "Insert a sleep after each step."
```

File: doc/source/plans.rst [hunk 3]
```diff
@@ -150,6 +181,46 @@ pseudo-axis. It's all the same to the plans. Examples:
...
+              yield from bluesky.plans.one_nd_step(detectors, step, pos_cache)
+              yield from bluesky.sleep(delay)
+
+          kwargs.setdefault('per_step', per_step)
```
Comment (mrakitin) [inline doc/source/plans.rst]: Shouldn't the second argument be `one_nd_step_with_delay`?
```suggestion
          kwargs.setdefault('per_step', one_nd_step_with_delay)
```

File: doc/source/plans.rst [hunk 4]
```diff
@@ -150,6 +181,46 @@ pseudo-axis. It's all the same to the plans. Examples:
...
+          yield from scan(*args, **kwargs)
```
Comment (mrakitin) [inline doc/source/plans.rst]:
```suggestion
          yield from bluesky.scan(*args, **kwargs)
```

File: doc/source/plans.rst [hunk 5]
```diff
@@ -109,6 +109,37 @@ Examples:
...
+   Or writing a custom plan that wraps :func:`count` and sets the exposure
+   time. This plan can encode the details that bluesky in general can't know.
+
+   .. code-block:: python
+
+      def count_with_time(detectors, num, delay, exposure_time, *, md=None):
+          # Assume all detectors have one exposure time component called
+          # 'exposure_time' that fully specifies its exposure.
+          for detector in detectors:
+              yield from bluesky.plans.mv(detector.exposure_time, exposure_time)
```
Comment (mrakitin) [inline doc/source/plans.rst]:
```suggestion
              yield from bluesky.plan_stubs.mv(detector.exposure_time, exposure_time)
```

File: doc/source/plans.rst [hunk 6]
```diff
@@ -150,6 +181,46 @@ pseudo-axis. It's all the same to the plans. Examples:
     RE.subscribe(bec)
     RE(scan([det], motor, 1, 5, 5))
 
+.. note::
+
+   Why don't scans have ``delay`` parameter?
```
Comment (teddyrendahl) [inline doc/source/plans.rst]:
```suggestion
   Why don't scans have a ``delay`` parameter?
```

File: doc/source/plans.rst (hunk 1 above)
Comment (danielballan) [reply doc/source/plans.rst]: Good question. I looked this up in [Garner's Modern American Usage](https://www.amazon.com/Garners-Modern-American-Usage-Garner/dp/0195382757) to check myself. He advises:

> In American English, the preferred form is *toward*; *towards* is prevalent in British English.

Comment (mrakitin) [reply doc/source/plans.rst]: Ah, got it! I learned British English at school 🇬🇧

File: doc/source/plans.rst [hunk 7]
```diff
@@ -109,6 +109,37 @@ Examples:
...
+
+   Additionally, counts with multiple detectors as in ``count([det1, det2]))``
+   would need to provide a separate exposure time for each detector in the
+   general case, which would grow wordy.
+
+   We recommend either setting the time-related parameter(s) in advance:
+
+   .. code-block:: python
+
+      det.exposure_time.set(3)
```
Comment (tacaswell) [inline doc/source/plans.rst]: I am wary of this code block because it will be copy-pasted into a plan.

File: doc/source/plans.rst [hunk 8]
```diff
@@ -109,6 +109,37 @@ Examples:
...
+   parameters. There is no one "exposure time" that can be applied to all
+   detectors.
+
+   Additionally, counts with multiple detectors as in ``count([det1, det2]))``
+   would need to provide a separate exposure time for each detector in the
+   general case, which would grow wordy.
+
+   We recommend either setting the time-related parameter(s) in advance:
+
+   .. code-block:: python
```
Comment (prjemian) [inline doc/source/plans.rst]: I agree with [Tom](https://github.com/bluesky/bluesky/pull/1223#pullrequestreview-255874096).  Suggest replacing lines 124-128 with this:

```
We recommend either setting the time-related parameter(s) in advance.
For interactive use:

   .. code-block:: python

      det.exposure_time.set(3)

From a plan:

   .. code-block:: python

      yield from bluesky.plan_stubs.mv(det.exposure_time, 3)
```

File: doc/source/plans.rst [hunk 9]
```diff
@@ -109,6 +109,37 @@ Examples:
...
     RE(count([noisy_det], num=5))
 
+.. note::
+
+   Why doesn't :func:`count` have an ``exposure_time`` parameter?
+
+   Modern CCD detectors typically parametrize exposure time with *multiple*
+   parameters. There is no one "exposure time" that can be applied to all
+   detectors.
+
```
Comment (prjemian) [inline doc/source/plans.rst]: Perfect place to expand on *why* a common method is a challenge.  Add this paragraph:

```
Scalers also parametrize exposure time with *multiple*
parameters.  The most common term (used by 
`EpicsScaler` and `ScalerCH`) is `det.preset_time`.
```

Do multi-channel analyzers and multi-channel scalers set the time differently.  This is the place to mention *briefly*.

**Comment by danielballan** [comment]
This is a surprising test failure:

```
=================================== FAILURES ===================================

[31m[1m________________________________ test_per_step _________________________________[0m



RE = <bluesky.run_engine.RunEngine object at 0x1248d6e48>

hw = namespace(ab_det=ABDetector(prefix='', name='det', read_attrs=['a', 'b'], configuration_attrs=[]), bool_sig=Signal(nam...Signal(name='sig', value=0, timestamp=1561990333.873343), trivial_flyer=<ophyd.sim.TrivialFlyer object at 0x126781518>)



[1m    def test_per_step(RE, hw):[0m

[1m        # Check default behavior, using one motor and then two.[0m

[1m        RE(scan([hw.det], hw.motor, -1, 1, 3, per_step=one_nd_step))[0m

[1m        RE(scan([hw.det],[0m

[1m                hw.motor, -1, 1,[0m

[1m                hw.motor2, -1, 1,[0m

[1m                3,[0m

[1m                per_step=one_nd_step))[0m

[1m        RE(inner_product_scan([hw.det], 3, hw.motor, -1, 1, per_step=one_nd_step))[0m

[1m        RE(inner_product_scan([hw.det],[0m

[1m                              3,[0m

[1m                              hw.motor, -1, 1,[0m

[1m                              hw.motor2, -1, 1,[0m

[1m                              per_step=one_nd_step))[0m

[1m    [0m

[1m        # Check that scan still accepts old one_1d_step signature:[0m

[1m        RE(scan([hw.det], hw.motor, -1, 1, 3, per_step=one_1d_step))[0m

[1m        RE(rel_scan([hw.det], hw.motor, -1, 1, 3, per_step=one_1d_step))[0m

[1m    [0m

[1m        # Test that various error paths include a useful error message identifying[0m

[1m        # that the problem is with 'per_step':[0m

[1m    [0m

[1m        # You can't usage one_1d_step signature with more than one motor.[0m

[1m        with pytest.raises(TypeError) as exc:[0m

[1m            RE(scan([hw.det],[0m

[1m                    hw.motor, -1, 1,[0m

[1m                    hw.motor2, -1, 1,[0m

[1m                    3,[0m

[1m                    per_step=one_1d_step))[0m

[1m>       assert "Signature of per_step assumes 1D trajectory" in str(exc)[0m

[1m[31mE       AssertionError: assert 'Signature of per_step assumes 1D trajectory' in '<ExceptionInfo TypeError tblen=7>'[0m

[1m[31mE        +  where '<ExceptionInfo TypeError tblen=7>' = str(<ExceptionInfo TypeError tblen=7>)[0m



[1m[31mbluesky/tests/test_new_examples.py[0m:676: AssertionError
```

**Comment by danielballan** [comment]
Hm, also seeing this failure on other recent PRs. Very strange.

**Comment by danielballan** [comment]
OK, problem identified and fixed in https://github.com/bluesky/bluesky/pull/1225. Unrelated to this change.

**Pull request merged**